app/models/deal.rb

```diff
@@ -51,6 +52,22 @@ def published?
     published_at.present?
   end
 
+  def expired?
+    expire_at < Time.current
+  end
+
+  def quantity_sold
+    orders.includes(:line_items).where(workflow_state: 'completed').sum('line_items.quantity')
```

Comment:
do we need includes? or can we use join?

Reply:
we can use both.